src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/BuildIntegratedRestoreUtility.cs

```diff
@@ -449,5 +449,63 @@ public static class BuildIntegratedRestoreUtility
 
             return closure;
         }
+
+        /// <summary>
+        /// Find the list of child projects direct or indirect references of target project in
+        /// reverse dependency order like the least dependent package first.
+        /// </summary>
+        public static async Task GetChildProjectsInClosure(BuildIntegratedNuGetProject target,
```

Comment:
I just tested it for perf, and there is not much difference. Also, I was able to work with UI during updation, it didn't lock it up. Having said that, it's better to cache these project references here since they will be used to get child references as well as parent references. So i've made this change to cache and it further improved perf by ~12 secs